src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -59,6 +69,63 @@ internal void InitializeLicense(string licenseText)
             license = foundLicense;
         }
 
+        static bool FirstTimeUser()
+        {
+            // Check for the presence of HKCU\Software\NServiceBus
+            using (var regNsbRoot = Registry.CurrentUser.OpenSubKey(@"Software\NServiceBus"))
+            {
+                if (regNsbRoot == null)
+                {
+                    //Check for the presence of HKCU\Software\ParticularSoftware
+                    using (var regParticularRoot = Registry.CurrentUser.OpenSubKey(@"Software\ParticularSoftware"))
+                    {
+                        if (regParticularRoot == null)
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        static async Task TrackFirstTimeUsageEvent()
+        {
+            // Get the current version of NServiceBus that's being used
+            var version = GitFlowVersion.MajorMinorPatch;
+
+            // Report first time usage metric
+            Logger.InfoFormat("Reporting first time usage and version information to www.particular.net. This call does not collect any personal information. For more details, see the License Agreement and the Privacy Policy available here: http://particular.net/licenseagreement. This call will NOT be executed in production servers. It is invoked only once when run in an interactive debugging mode when the endpoint is executed for the very first time.");
+            const string webApiUrl = "https://particular.net/api/ReportFirstTimeUsage";
+            var postData = $"version={version}";
+            try
+            {
+                using (var httpClient = new HttpClient())
```

Comment:
@indualagarsamy expressed some concerns about the using approach. It looks good to me, but we should make sure we're all aligned before merging this.